### DISCUSSION TITLE
to: run a shell script and paste the output at cursor location

### DISCUSSION THREAD

**Discussion by cova-fe** [pqrs-org/Karabiner-Elements#3822] [Q&A]
Hi all, 
I recently started to use Karabiner and I'm still exploring his capabilities (amazing tool, btw). Now I would like to know if it is possible to run a shell script AND paste the output at cursor position. I'm already using the "command" option for "to:", but now I'm wondering if it is possible to capture the output. (example use case: Authenticator otp creation from command line)
Many thanks for any answer.

**Comment by tekezo** [comment]
You can use the pbcopy command.
The key point is to specify `hold_down_milliseconds` to wait for the shell execution before pasting.

```json
{
            "to": [
                {
                    "hold_down_milliseconds": 500,
                    "shell_command": "date | pbcopy"
                },
                {
                    "key_code": "v",
                    "modifiers": [
                        "left_command"
                    ]
                }
            ]
}
```

**Reply by cova-fe** [reply]
Good hint indeed, and it almost worked :) . I had to implement it in a slightly different way, because I'm using a yubikey and a fixed delay is not the best option. I opted to write a small go helper that puts directly in the otp in the clipboard and then I manually paste the result. Not the perfect result but good enough. I understand that karabiner has no way to wait for a shell command to return, right?
